docs/addons/karpenter.md

```diff
@@ -0,0 +1,146 @@
+# Karpenter Add-on
+
+Karpenter add-on is based on the [Karpenter](https://github.com/aws/karpenter) open source node provisioning project. It provides a more efficient and cost-effective way to manage workloads by launching just the right compute resources to handle a cluster's application. 
+
+Karpenter works by:
+
+* Watching for pods that the Kubernetes scheduler has marked as unschedulable,
+* Evaluating scheduling constraints (resource requests, nodeselectors, affinities, tolerations, and topology spread constraints) requested by the pods,
+* Provisioning nodes that meet the requirements of the pods,
+* Scheduling the pods to run on the new nodes, and
+* Removing the nodes when the nodes are no longer needed
+
+## Prerequisite
+
+Here are the prerequisites:
+1. Karpenter discovers subnets tagged `kubernetes.io/cluster/$CLUSTER_NAME`. Private subnets where the EKS cluster is deployed shoule be tagged as such.
```

Comment:
should note that the framework take care of this tag.

Reply:
moved to the functionality section.